src/main/java/net/rubygrapefruit/platform/internal/jni/AbstractFileEventFunctions.java

```diff
@@ -66,33 +68,43 @@ public void reportError(Throwable ex) {
         }
     }
 
-    // Instantiated from native code
-    @SuppressWarnings("unused")
     protected static class NativeFileWatcher implements FileWatcher {
-        /**
-         * A Java object wrapper around the native server object.
-         */
-        private Object server;
+        private final Object server;
+        private final Thread processorThread;
+        private boolean closed;
 
-        public NativeFileWatcher(Object server) {
+        public NativeFileWatcher(final Object server) throws InterruptedException {
             this.server = server;
+            final CountDownLatch runLoopInitialized = new CountDownLatch(1);
+            this.processorThread = new Thread("File watcher server") {
+                @Override
+                public void run() {
+                    initializeRunLoop0(server);
+                    runLoopInitialized.countDown();
+                    executeRunLoop0(server);
+                }
+            };
+            this.processorThread.setDaemon(true);
+            this.processorThread.start();
+            // TODO Parametrize this
+            runLoopInitialized.await(5, TimeUnit.SECONDS);
```

Comment:
You need to check the return value here. The `await` method doesn't throw an exception when it times out, but returns `false` instead.

Reply:
You're right. This API is so confusing. At least it should be annotated with `@CheckReturnValue`.